src/wwwroot/app/services/VacancyService.js

```diff
@@ -134,14 +137,36 @@ export default class VacancyService {
       let userPromise = _VacancyService._getUser(vacancy);
       let thesaurusesPromises = _VacancyService._getThesauruses(vacancy);
       let childVacancyPromises = _VacancyService._getChildVacancies(vacancy);
+      let commentsPromise = _VacancyService._getCommentsFields(vacancy);
       let closingCandidatePromise = _VacancyService._getClosingCandidate(vacancy);
-      return _$q.all([userPromise, thesaurusesPromises, childVacancyPromises, closingCandidatePromise]);
+      return _$q.all([userPromise, thesaurusesPromises, childVacancyPromises,
+             commentsPromise, closingCandidatePromise]);
    }
 
    _getUser(vacancy) {
       return _UserService.getUserById(vacancy.responsibleId);
    }
 
+   _getCommentsFields(vacancy) {
+      if (vacancy.comments) {
+         let promises = map(vacancy.comments, (comment) => {
+            comment.createdOn = utils.formatDateFromServer(comment.createdOn);
+            _UserService.getUserById(comment.authorId).then(user => set(comment, 'responsible', user));
+            return comment;
+         });
+         return _$q.all(promises);
+      }
+   }
+
+   _convertCommentsToServer(vacancy) {
```

Comment:
No, this method should convert date and remove property 'responsible' from comment